locallib.php

```diff
@@ -2246,18 +2246,370 @@ function bigbluebuttonbn_output_recording_table($bbbsession, $recordings, $tools
+            }
+        }
+        if (!empty($imageurl)) {
+            $recordingpreview .= bigbluebuttonbn_get_opencast_recording_data_row_preview_images($imageurl);
+        }
+    }
+   
+    $recordingpreview .= html_writer::end_tag('div');
+    return $recordingpreview;
+}
+
+/**
+ * Helper function format Opencast recording date used in row for the data used by the recording table.
+ *
+ * @param array $ocrecording
+ *
+ * @return string
+ */
+function bigbluebuttonbn_get_opencast_recording_data_row_date_formatted($ocrecording) {
+    global $USER;
+    $starttime_str = !empty($ocrecording['start']) ? $ocrecording['start'] : $ocrecording['created'];
+    $starttime = strtotime($starttime_str);
+    // Set formatted date.
+    $dateformat = get_string('strftimerecentfull', 'langconfig') . ' %Z';
+    return userdate($starttime, $dateformat, usertimezone($USER->timezone));
+}
+
+/**
+ * Helper function converts Opencast recording duration used in row for the data used by the recording table.
+ *
+ * @param array $duration
+ *
+ * @return integer
+ */
+function bigbluebuttonbn_get_opencast_recording_data_row_duration($duration) {
+    if ($duration) {
+        // Convert the duration (in miliseconds) into Hours:Minutes:Seconds format
+        return gmdate('H:i:s', $duration / 1000);
+    }
+    return 0;
+}
+
+/**
+ * Helper function builds Opencast recording actionbar used in row for the data used by the recording table.
+ *
+ * @param array $ocrecording
+ * @param array $bbbsession
+ *
+ * @return string
+ */
+function bigbluebuttonbn_get_opencast_recording_data_row_actionbar($ocrecording, $bbbsession) {
+    global $OUTPUT;
+    if (empty($ocrecording['identifier']) || empty($bbbsession['course'])) {
+        return '';
+    }
+    $actionbar = '';
+    $linkattributes = array(
+        'target' => '_blank',
+        'class' => 'btn btn-xs btn-danger'
+    );
+    // Creating moodle url, to redirect to Opencast update metadata (Edit) page.
+    $opencastediturl = new \moodle_url('/blocks/opencast/updatemetadata.php',
+            array('video_identifier' => $ocrecording['identifier'], 'courseid' => $bbbsession['course']->id));
+    $linkattributes['id'] = 'opencast-edit-episode-' . $ocrecording['identifier'];
+    // Generating Action Link for Opencast update metadata (Edit).
+    $actionbar .= $OUTPUT->action_link($opencastediturl, get_string('edit'), null, $linkattributes) . '&#32;';
+    
+    // Creating moodle url, to redirect to Opencast delete event (Delete) page.
+    $opencastdeleteurl = new \moodle_url('/blocks/opencast/deleteevent.php',
+            array('identifier' => $ocrecording['identifier'], 'courseid' => $bbbsession['course']->id));
+    $linkattributes['id'] = 'opencast-delete-episode-' . $ocrecording['identifier'];
+    // Generating Action Link for Opencast delete event (Delete).
+    $actionbar .= $OUTPUT->action_link($opencastdeleteurl, get_string('delete'), null, $linkattributes) . '&#32;';
```

Comment:
As commented above, I prefer not to extend and make changes to other plugins only for this integration, in order to avoid complications. Therefore, if the current behavior is not acceptable then we do not offer it.